### DISCUSSION TITLE
Расшариваемая библиотека шаблонов

### DISCUSSION THREAD

**Discussion by simplepersonru** [totumonline/totum-ru-issues-and-discussions#416] [Как это сделать?] [answered]
Иногда бывает что одно и то же поле или группа полей используется в нескольких таблицах. Например - "ответственный сотрудник". Или "наблюдатели". Это примеры которые вероятно много у кого будут такие, но их придумать можно больше

Хочется чтобы была возможность сохранять поля таблицы со всеми их настройками, кодами и тд в свою библиотеку шаблонов. И позже применять эти поля на какие-то свои таблицы. Работало бы также, как с добавлением поля, но настройки, коды и тд предзаполнены из шаблона

Если развить идею, то можно делать эти библиотеки расшариваемые, чтобы можно было делиться друг с другом наработками. Например пускай тотум сервер будет уметь по определенному порту отдавать json со всеми шаблонами (делимся своими наработками). При этом можно ввести ряд подключений, откуда бы мы импортировали шаблоны (пользуемся наработками других). 
Либо просто экспортировать в файл и делиться на форуме, наверное это проще

Если еще развить идею, то можно сделать вот такие отдельные секции для шаблонов:
1. Коды
2. Коды действий
3. Коды форматирования
4. Коды селектов
5. Поля таблицы(со всеми настройками и кодами)
6. Таблицы целиком

**Comment by erpic-ru** [comment]
Есть стандартный механизм импорта и экспорта таблиц плюс возможность дублирования таблиц в списке, в том числе и расчетных в цикле. К тому же настроить несколько полей в тотуме, порой, быстрее, чем заниматься  импортом и экспортом (пусть даже автоматическим).

Что касается шаринга – можете начать с себя: выложить на форуме какие-то шаблонные штуки, например, интеграции со сторонними сервисами.

**Reply by simplepersonru** [reply]
Но ведь не получится просто какое-то поле одно импортировать. Только таблицу целиком. Далеко не всегда этого достаточно

**Comment by ttm-alexeypolunin** [comment]
Сколько вы готовы за это заплатить?

**Reply by simplepersonru** [reply]
заплатить в смысле за:

1. Себестоимость такой разработки?
Думаю на это у меня не хватит денег, либо нужно расчехлять запасы, измеряется сотнями тысяч как минимум. Я обычный работяга программист со средней программистской зарплатой

2. Приоритезацию фичи для следующих релизов?
Прямо сейчас готов вывалить 15к рублей с небольшим предварительным обсуждением фичи. Если бы тотум прямо сейчас приносил мне деньги (что в скором времени будет, но пока только движение в эту сторону), был бы готов расчехлить 50-100к рублей, в зависимости от согласованного объема функционала. 

Я понимаю, что это смешные деньги для себестоимости коммерческой разработки. Думаю, что для покрытия затрат на разработку, здесь нужен механизм аля кикстартер, где инвесторами будут выступать те, кто активно используют тотум для получения прибыли

Я рассуждаю следующим образом - есть форум и есть топик для предложения идей. Вот я и предлагаю, авось вызову у мейнтейнеров резонанс и они загорятся сделать это со страшной силой :D :D. Это не то чтобы киллер фича - она позволяет экономить время и создавать свои библиотеки переиспользуемых "контролов" из которых можно чрезвычайно быстро стряпать решение или расширять существующее. Но ничего такого, чего нельзя было бы сделать сейчас (просто может подольше).

**Reply by ttm-alexeypolunin** [reply, marked as answer]
Да, где-то 150-200 на такое уйдет. Весь Тотум стоил 30М и вроде в соотношении как-бы немного, но я ту сменил обстановку за окном и понял, какой ужасной фигней я занимаюсь делая бесплатный продукт. Что не тоже самое, что open source проект. В какой-то момент у меня эти два пункта склеились вместе, хотя open source и законченный продукт с саппортом и прочими ништяками это кардинально про разное.

Были случаи, когда за какой-то модуль, который долго стоял у нас в планах платили 100-150К, но это скорее исключение. В основном обратные случаи, когда люди делают проект за 5-10 млн на Тотум и не хотят 5 тр заплатить за консультацию тк в их картине мира мы им должны. Были случаи, когда люди донатили 5 тр, а потом требовали, что бы мы их консультировали бессрочно и безлимитно. Были компании у которых заводская территория измеряется тысячами гектар, которые мне в телеграм вереницу выпрашивателей бесплатного PRO отправляли. Все это очень увлекательно, но немного не про то, что нужно.

Поэтому написать то предложение можно, но делать мы его не будем — все по честному.

По этой же причине больше нет бесплатного саппорта в telegram. По этой же причине я отвечаю на этом форуме максимум раз в неделю.

Для того, что бы Totum продолжил существование — мы далее занимаемся только оплаченной разработкой.

**Reply by simplepersonru** [reply]
> в их картине мира мы им должны

Я очень благодарен за такой продукт, как только смогу на этом заработать, непременно поделюсь)

> Поэтому написать то предложение можно, но делать мы его не будем — все по честному.

Хозяин - барин, но в таком случае не вполне понимаю цель существования этого топика) 
Упс. Только что заметил что написал не в "Есть идея", а "Как это сделать" - моя ошибка.

> Для того, что бы Totum продолжил существование — мы далее занимаемся только оплаченной разработкой.

С этим сложно поспорить

**Reply by ttm-alexeypolunin** [reply]
Это я перевел в другой топик. Суть «есть идея» — утрачена. Надо с ним что-то сделать, удалить или закенселить.
